src/js/components/modals/AppModalComponent.jsx

```diff
@@ -29,27 +28,32 @@ var AppModalComponent = React.createClass({
   displayName: "AppModalComponent",
 
   propTypes: {
+    app: React.PropTypes.object,
     attributes: React.PropTypes.object,
-    edit: React.PropTypes.bool,
     onDestroy: React.PropTypes.func
   },
 
   getDefaultProps: function () {
     return {
+      app: null,
       attributes: lazy(appScheme).extend({
         cpus: 0.1,
         instances: 1,
         mem: 16.0,
         disk: 0.0
       }).value(),
-      edit: false,
       onDestroy: Util.noop
     };
   },
 
   getInitialState: function () {
+    var app = this.props.app;
     AppFormStore.initAndReset();
 
+    if (app != null) {
+      AppFormStore.populateFieldsFromAppDefinition(app);
```

Comment:
For the records.
Works if I set the state then with AppFormStore.fields, which would introduce an additional render.